src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/AssessmentStatus.kt

```diff
@@ -393,21 +396,35 @@ sealed interface SideEffect {
 }
 
 sealed class AssessmentLifecycleEvent {
-  data class EligibilityAndSuitabilityAnswerProvided(val eligibilityStatus: EligibilityStatus) : AssessmentLifecycleEvent()
-  data class ResidentialCheckStatusAnswerProvided(val checkStatus: ResidentialChecksStatus) : AssessmentLifecycleEvent()
+  data class EligibilityAndSuitabilityAnswerProvided(val eligibilityStatus: EligibilityStatus, val answer: CriterionCheck) : AssessmentLifecycleEvent() {
+    override fun getContext(): Map<String, Any> = mapOf("eligibilityStatus" to eligibilityStatus, "type" to answer.type, "code" to answer.code, "answers" to answer.answers)
+  }
+  data class ResidentialCheckStatusAnswerProvided(val checkStatus: ResidentialChecksStatus, val answer: SaveResidentialChecksTaskAnswersRequest) : AssessmentLifecycleEvent() {
```

Comment:
For these can we pass the taskCode and answers in separately? 
Just so the persistence layer is relying on the web layer